spec/reservation_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe Reservation, type: :model do
+  before(:all) do
+    @user = User.create(username: "john doe1#{SecureRandom.hex}", password: '123456', password_confirmation: '123456')
+    expect(@user).to be_valid
```

Comment:
This can work fine, but it's not common to see tests in the life-cycle hooks such as `before(:all)`. I think it would be better if it was in a separate test.